src/handlers/messages-query.ts

```diff
@@ -0,0 +1,67 @@
+import type { DidResolver } from '@web5/dids';
+import type { EventLog } from '../types/event-log.js';
+import type { MessageStore } from '../types/message-store.js';
+import type { MethodHandler } from '../types/method-handler.js';
+import type { MessagesQueryMessage, MessagesQueryReply } from '../types/messages-types.js';
+
+import { authenticate } from '../core/auth.js';
+import { messageReplyFromError } from '../core/message-reply.js';
+import { Messages } from '../utils/messages.js';
+import { MessagesGrantAuthorization } from '../core/messages-grant-authorization.js';
+import { MessagesQuery } from '../interfaces/messages-query.js';
+import { PermissionsProtocol } from '../protocols/permissions.js';
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
+
+
+export class MessagesQueryHandler implements MethodHandler {
+
+  constructor(private didResolver: DidResolver, private messageStore: MessageStore, private eventLog: EventLog) { }
+
+  public async handle({
+    tenant,
+    message
+  }: {tenant: string, message: MessagesQueryMessage}): Promise<MessagesQueryReply> {
+    let messagesQuery: MessagesQuery;
+
+    try {
+      messagesQuery = await MessagesQuery.parse(message);
+    } catch (e) {
+      return messageReplyFromError(e, 400);
+    }
+
+    try {
+      await authenticate(message.authorization, this.didResolver);
+      await MessagesQueryHandler.authorizeMessagesQuery(tenant, messagesQuery, this.messageStore);
+    } catch (e) {
+      return messageReplyFromError(e, 401);
+    }
+
+    // an empty array of filters means no filtering and all events are returned
+    const eventFilters = Messages.convertFilters(message.descriptor.filters);
+    const { events, cursor } = await this.eventLog.queryEvents(tenant, eventFilters, message.descriptor.cursor);
```

Comment:
Mainly for setting expectation, would this be changed to something like below in the future?

```suggestion
    const { messages, cursor } = await this.messageLog.queryMessages(tenant, messageFilters, message.descriptor.cursor);
```